church_of_jesus_christ_api/church_of_jesus_christ_api.py

```diff
@@ -1,523 +1,550 @@
+        endpoint = _endpoints[name]
+        def default_if_none(val, default):
+            return str(val if val != None else default)
+
+        if self.__user_details:
+            endpoint = endpoint.replace('{unit}', default_if_none(unit, self.__user_details['homeUnits'][0]))
+            endpoint = endpoint.replace('{member_id}', default_if_none(member_id, self.__user_details['individualId']))
+            endpoint = endpoint.replace('{uuid}', default_if_none(uuid, self.__user_details['uuid']))
+        return endpoint
+
+    def __get_JSON(self, endpoint: str, headers: Dict[str,str] = None) -> JSONType:
+        return self.__session.get(
+                endpoint, 
+                headers=headers, 
+                timeout=15, 
+                verify=self.__verify_SSL).json()
+    
+    @property
+    def session(self):
+        '''
+        Returns the requests session being used by the API.
+        '''
+        return self.__session
+    
+    @property
+    def user_details(self):
+        '''
+        Returns the details of the user logged into this session
+
+        Returns
+        
+        .. literalinclude:: ../JSON_schemas/user_details-schema.md
+        '''
+        return self.__user_details
+
+    def download_patriarchal_blessing(self, filename: str = 'patriarchal_blessing.pdf') -> None:
+        '''
+        Downloads user's patriarchal blessing (if available online) and saves it to a file with the given name
+
+        Parameters
+        
+        filename : str
+            The name of the file in which to save the PDF of the patriarchal blessing 
+        '''
+
+        file_content = self.__session.get(self.__endpoint('patriarchal-blessing'), verify=self.__verify_SSL, timeout=15).content
+        return open(filename, 'wb').write(file_content)
+
+    def get_action_and_interviews(self, unit: int = None) -> JSONType:
+        '''
+        Returns the unit action and interview list
+
+        Parameters
+        
+        unit : int
+            Number of the church unit for which to retrieve the report
+
+        Returns
+        
+        .. literalinclude:: ../JSON_schemas/get_action_and_interviews-schema.md
+        '''
+
+        return self.__get_JSON(self.__endpoint('action-and-interviews', unit=unit))
+
+    def get_attendance(self, start_date: datetime.date = None, end_date: datetime.date = None, unit: int = None) -> JSONType:
+        '''
+        Returns the attendance list
+
+        Parameters
+
+        start_date: datetime.date
+            The start date after which donations will be retrieved
+        end_date: datetime.date
+            The end date after which donations will no longer be retrieved
+        unit : int
+            Number of the church unit for which to retrieve the report
+
+        Returns
+
+        .. literalinclude:: ../JSON_schemas/get_attendance-schema.md
+        '''
+
+        def convert_date_to_string_using_default_date_if_none(val, default):
```

Comment:
Now that this is being used in two different methods, make it either a private member method or private free function in the module